samtranslator/model/sam_resources.py

```diff
@@ -2237,6 +2245,94 @@ def _construct_appsync_api_resources(
 
         return api, cloudwatch_role
 
+    def _parse_auth_properties(self, api: GraphQLApi, auth: aws_serverless_graphqlapi.Auth) -> None:
+        """
+        Parse the Auth properties in a Serverless::GraphQLApi resource.
+
+        This function does not return a value. The GraphQLApi object is modified directly so that we don't
+        return multiple config properties which are mostly None values.
+        """
+        # Default authentication
+        name, auth_dict = self._validate_auth_type_and_config(auth)
+        api.AuthenticationType = auth.Type
+        if name:
+            setattr(api, name, auth_dict)
```

Comment:
Why `setattr`?

Reply:
Feels like the best approach since we have multiple types of configs that can be returned, and this makes the function really nice to use since it works for dictionaries too where name is key and auth_dict is value (this is how its done for AddtionalAuthenticationProviders).

Reply:
Feels like the best approach since we have multiple types of configs that can be returned, and this makes the function really nice to use since it works for dictionaries too where name is key and auth_dict is value (this is how its done for AddtionalAuthenticationProviders).

Reply:
Perhaps, but I'm not a big fan of `setattr` when it can avoided, since it has wide potential impact if there's any bugs (now or in the future) in the code before it. See e.g. https://snyk.io/blog/the-dangers-of-setattr-avoiding-mass-assignment/

Reply:
I would see more potential issue if our `name` value was being parsed from some user input or it was variably generated, but these are hardcoded values that have no way of causing this level of impact.

Reply:
Would the alternative (i.e. without `setattr`) be a lot uglier or more complicated? And while it's true there may not be any issue right now, it's good to minimize/eliminate categories of bugs even down the road as the code changes.

Reply:
Conclusion: After some offline discussion, we decided we will keep `setattr` for now. Accessing top level properties in resource is by class attribute (like `api.Auth`), but nested properties like `api.AdditionalAuth...` are dictionaries which results in this mixed approach for how we set values. Left a TODO for us to look into changing `Resource` so that all the properties are stored in dictionaries to avoid future issues like this.